test/ejabberd_tests/tests/ejabberdctl_SUITE.erl

```diff
@@ -595,10 +608,175 @@ stats_host(Config) ->
                 {"0\n", 0} = ejabberdctl("stats_host", ["onlineusers", SecDomain], Config)
         end).
 
+
+
+%%-----------------------------------------------------------------
+%% Improve coverage
+%%-----------------------------------------------------------------
+
+
+
+simple_register(Config) ->
+    %% given
+    Domain = escalus_ct:get_config(ejabberd_domain),
+    {Name, Password} = {<<"tyler">>, <<"durden">>},
+    %% when
+    {_, _} = ejabberdctl("register", [Name, Domain, Password], Config),
+    {R2, _} = ejabberdctl("registered_users", [Domain], Config),
+    %% then
+    {match, _} = re:run(R2, ".*(" ++binary_to_list(Name)++").*").
+
+simple_unregister(Config) ->
+    %% given
+    Domain = escalus_ct:get_config(ejabberd_domain),
+    {Name, _} = {<<"tyler">>, <<"durden">>},
+    %% when
+    {_, _} = ejabberdctl("unregister", [Name, Domain], Config),
+    {R2, _} = ejabberdctl("registered_users", [Domain], Config),
+    %% then
+    nomatch = re:run(R2, ".*(" ++binary_to_list(Name)++").*").
+
+register_twice(Config) ->
+    %% given
+    Domain = escalus_ct:get_config(ejabberd_domain),
+    {Name,  Password} = {<<"tyler">>, <<"durden">>},
+    %% when
+    {_, _} = ejabberdctl("register", [Name, Domain, Password], Config),
+    {R, _} = ejabberdctl("register", [Name, Domain, Password], Config),
+    %% then
+    {match, _} = re:run(R, ".*(already registered).*"),
+    {_, _} = ejabberdctl("unregister", [Name, Domain], Config).
+
+
+
+backup_restore_mnesia(Config) ->
+    %% given
+    TableName = passwd,
+    TableSize = rpc_call(mnesia, table_info, [TableName, size]),
+    io:format("Table size is ~n~p~n", [TableSize]),
+    %% Table passwd should not be empty
+    FileName = "backup_mnesia.bup",
+    %% when
+    {R, _} = ejabberdctl("backup", [FileName], Config),
+    nomatch = re:run(R, ".+"),
+    rpc_call(mnesia, clear_table, [TableName]),
+    0 = rpc_call(mnesia, table_info, [TableName, size]),
+    {R2, _} = ejabberdctl("restore", [FileName], Config),
+    %% then
+    nomatch = re:run(R2, ".+"),
+    TableSize = rpc_call(mnesia, table_info, [TableName, size]).
+
+restore_mnesia_wrong(Config) ->
+    FileName = "file that doesnt exist13123.bup",
+    {R2, _} = ejabberdctl("restore", [FileName], Config),
+    {match, _} = re:run(R2, ".+").
+
+dump_and_load(Config) ->
```

Comment:
In thi case,  they differ in output file format. `Dump` writes database tables to a text file (erlang syntax) . `Backup` ends with raw (called "opaque")  file, readable by mnesia. It performs also some other other operations (activte/deactivate checkpoints) in background and then allows you to execute some operations on it e.g traverse over.